pkg/stanza/operator/input/windows/input.go

```diff
@@ -42,28 +106,54 @@ func (i *Input) Start(persister operator.Persister) error {
 
 	i.persister = persister
 
+	if i.isRemote() {
+		if err := i.startRemoteSession(); err != nil {
+			i.Logger().Error("Failed to start remote session", zap.Error(err))
+			return fmt.Errorf("failed to start remote session: %w", err)
+		}
+	}
+
 	i.bookmark = NewBookmark()
 	offsetXML, err := i.getBookmarkOffset(ctx)
 	if err != nil {
-		i.Logger().Error("Failed to open bookmark, continuing without previous bookmark", zap.Error(err))
+		i.Logger().Error("Failed to get bookmark offset", zap.Error(err))
 		_ = i.persister.Delete(ctx, i.channel)
 	}
 
 	if offsetXML != "" {
 		if err := i.bookmark.Open(offsetXML); err != nil {
+			i.Logger().Error("Failed to open bookmark", zap.Error(err))
 			return fmt.Errorf("failed to open bookmark: %w", err)
 		}
 	}
 
-	i.subscription = NewSubscription()
-	if err := i.subscription.Open(i.channel, i.startAt, i.bookmark); err != nil {
-		return fmt.Errorf("failed to open subscription: %w", err)
+	i.publisherCache = newPublisherCache()
+
+	subscription := NewLocalSubscription()
+	if i.isRemote() {
+		subscription = NewRemoteSubscription(i.remote.Server)
 	}
 
-	i.publisherCache = newPublisherCache()
+	if err := subscription.Open(i.channel, i.startAt, i.bookmark, uintptr(i.remoteSessionHandle)); err != nil {
+		if isNonTransientError(err) {
+			if i.isRemote() {
+				i.Logger().Error("Failed to open subscription for remote server", zap.String("server", i.remote.Server), zap.Error(err))
```

Comment:
Isn't this going to generate two log lines with very similar content? At first I prefer to include the server name of the `fmt.Errorf` message and let the host log it.

Reply:
yeah, updated.